reasoning/src/jvmMain/kotlin/com/xebia/functional/xef/reasoning/pdf/ReadPDFFromUrl.kt

```diff
@@ -28,6 +33,12 @@ constructor(
 
   override val description: String = "Reads the content of a PDF as String from a URL"
 
+  private val coroutineScope = CoroutineScope(SupervisorJob())
```

Comment:
In some cases is convenient in services like this one to declare a `coroutineScope` that allows turning `suspend` APIs and offering a version for `CompletableFuture`. In this case, we do it like this because the PDF functionality is only available in the JVM at the moment of writing this.